lib/matplotlib/axes/_base.py

```diff
@@ -417,10 +417,16 @@ def _plot_args(self, tup, kwargs, return_kwargs=False):
             raise ValueError("x, y, and format string must not be None")
 
         kw = {}
-        for k, v in zip(('linestyle', 'marker', 'color'),
-                        (linestyle, marker, color)):
+        for prop_name, v in zip(('linestyle', 'marker', 'color'),
+                                (linestyle, marker, color)):
             if v is not None:
-                kw[k] = v
+                if prop_name in kwargs:
+                    _api.warn_external(
+                        f"'{prop_name}' is redundantly defined via fmt string "
```

Comment:
A lot of users are not going to know what `fmt` refers to, particularly those who will put the information in two places.  Can we be more explicit here?
```
linestyle is redundantly defined by the "linestyle" keyword argument and the fmt string ("--r").  The keyword argument will take precedence.
```

I think that requires saving `tup[-1]`, but that isn't too hard.

Reply:
Done. Took the opportunity to clean up the function (#19277) and not make it even more confusing by saving a bogous `tup[-1]`.

Reply:
#19278...